tests/test_bgp_af.rb

```diff
@@ -828,7 +1117,15 @@ def network_cmd(af, dbg)
                  "#{dbg} Test 3. Restore the removed networks")
 
     # Test: Change route-maps on existing networks
-    should = master.map { |network, rm| [network, "#{rm}_55"] }
+    if platform == :ios_xr
+      %w(rtmap1_55, rtmap2_55, rtmap3_55, rtmap5_55,
+         rtmap6_55, rtmap7_55).each do |policy|
+        config("route-policy #{policy}", 'end')
+      end
+      should = master.map { |network, rm| [network, rm.nil? ? nil : "#{rm}_55"] }
```

Comment:
without the nil check test_network fails with:

D, [2015-12-02T11:00:59.117090 #5537] DEBUG -- : CLI sent to device:
D, [2015-12-02T11:00:59.117138 #5537] DEBUG -- : router bgp 1
D, [2015-12-02T11:00:59.117177 #5537] DEBUG -- :  address-family ipv4 unicast
D, [2015-12-02T11:00:59.117213 #5537] DEBUG -- :   network 192.168.8.0/24 route-policy _55
D, [2015-12-02T11:00:59.117285 #5537] DEBUG -- : Sending 'cli_config' request with args: '#<IOSXRExtensibleManagabilityService::CliConfigArgs:0x00000001347dc0>'
D, [2015-12-02T11:00:59.252547 #5537] DEBUG -- : Handling 'cli_config' reply:
#<IOSXRExtensibleManagabilityService::CliConfigReply:0x00000001346c18>
D, [2015-12-02T11:00:59.252659 #5537] DEBUG -- : Reply includes errors:
{
 "cisco-grpc:errors": {
  "error": [
   {
    "error-type": "application",
    "error-tag": "operation-failed",
    "error-severity": "error",
    "error-message": "\nUnable to process cmd, ret-val: 3, cmd:   network 192.168.8.0/24 route-policy _55"
   }
  ]
 }
}

TestRouterBgpAF#test_network = 35.67 s = E

Finished in 35.669107s, 0.0280 runs/s, 0.0841 assertions/s.

  1) Error:
TestRouterBgpAF#test_network:
Cisco::CliError: CliError: 'input TODO' rejected with message:
'["\nUnable to process cmd, ret-val: 3, cmd:   network 192.168.8.0/24 route-policy _55"]'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/node.rb:271:in `rescue in config'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/node.rb:263:in `config'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/node.rb:162:in `config_set'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/node_util.rb:66:in `config_set'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/bgp_af.rb:557:in `block (2 levels) in networks='
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/bgp_af.rb:551:in `each'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/bgp_af.rb:551:in `block in networks='
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/bgp_af.rb:548:in `each'
    /home/gvlaemyn/cisco-network-node-utils/lib/cisco_node_utils/bgp_af.rb:548:in `networks='
    test_bgp_af.rb:1131:in `network_cmd'
    test_bgp_af.rb:1063:in `block (2 levels) in test_network'
    test_bgp_af.rb:1059:in `each'
    test_bgp_af.rb:1059:in `block in test_network'
    test_bgp_af.rb:1058:in `each'
    test_bgp_af.rb:1058:in `test_network’

> On 30 Nov 2015, at 22:52, Glenn Matthews <notifications@github.com> wrote:
> 
> In tests/test_bgp_af.rb <https://github.com/cisco/cisco-network-node-utils/pull/87#discussion_r46209232>:
> 
> > @@ -828,7 +1117,15 @@ def network_cmd(af, dbg)
> >                   "#{dbg} Test 3. Restore the removed networks")
> >  
> >      # Test: Change route-maps on existing networks
> > -    should = master.map { |network, rm| [network, "#{rm}_55"] }
> > +    if platform == :ios_xr
> > +      %w(rtmap1_55, rtmap2_55, rtmap3_55, rtmap5_55,
> > +         rtmap6_55, rtmap7_55).each do |policy|
> > +        config("route-policy #{policy}", 'end')
> > +      end
> > +      should = master.map { |network, rm| [network, rm.nil? ? nil : "#{rm}_55"] }
> Curious why you need a nil check for XR but not for NXOS?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/cisco/cisco-network-node-utils/pull/87/files#r46209232>.
>